internal/consensus/state.go

```diff
@@ -1560,10 +1613,12 @@ func (cs *State) enterPrevoteWait(height int64, round int32) {
 	logger := cs.Logger.With("height", height, "round", round)
 
 	if cs.Height != height || round < cs.Round || (cs.Round == round && cstypes.RoundStepPrevoteWait <= cs.Step) {
-		logger.Debug(
-			"entering prevote wait step with invalid args",
-			"current", log.NewLazySprintf("%v/%v/%v", cs.Height, cs.Round, cs.Step),
-		)
+		if logger.DebugOn() {
+			logger.Debug(
+				"entering prevote wait step with invalid args",
+				"current", log.NewLazySprintf("%v/%v/%v", cs.Height, cs.Round, cs.Step),
```

Comment:
We don't need the `log.NewLazySprintf` in this case, right? The idea of this feature was to avoid paying the cost of printing when we are not printing because the logger level is not the desired one (debug).